File: src/main/java/com/camackenzie/exvi/core/ExerciseMechanics.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core;

/**
 *
 * @author callum
 */
public enum ExerciseMechanics {
    ISOLATION,
    COMPOUND;

    public static ExerciseMechanics fromString(String s) {
        return EnumUtils.enumFromString(ExerciseMechanics.class, s);
    }

    @Override
    public String toString() {
        return EnumUtils.formatName(super.toString());
    }
}

File: common/src/commonMain/kotlin/com/camackenzie/exvi/core/model/ExerciseMechanics.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model

import com.camackenzie.exvi.core.model.EnumUtils.formatName

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
enum class ExerciseMechanics {
    ISOLATION, COMPOUND, OTHER;

    override fun toString(): String {
        return formatName(super.toString())
    }

    companion object {
        fun fromString(s: String): ExerciseMechanics? {
            return EnumUtils.enumFromString<ExerciseMechanics>(s)
        }
    }
}